pkg/cvo/internal/operatorstatus.go

```diff
@@ -167,48 +166,71 @@ func waitForOperatorStatusToBeDone(ctx context.Context, interval time.Duration,
+		degraded_value := true
+		var degradedCondition *configv1.ClusterOperatorStatusCondition
+		for i := range actual.Status.Conditions {
+			condition := &actual.Status.Conditions[i]
 			switch {
 			case condition.Type == configv1.OperatorAvailable && condition.Status == configv1.ConditionTrue:
 				available = true
 			case condition.Type == configv1.OperatorProgressing && condition.Status == configv1.ConditionFalse:
 				progressing = false
-			case condition.Type == configv1.OperatorFailing && condition.Status == configv1.ConditionFalse:
-				failing = false
+			case condition.Type == configv1.OperatorFailing:
+				if condition.Status == configv1.ConditionFalse {
+					failing = false
+				}
+				failingCondition = condition
+			case condition.Type == configv1.OperatorDegraded:
+				if condition.Status == configv1.ConditionFalse {
+					degraded_value = false
+				}
+				degradedCondition = condition
 			}
 		}
+
+		// If degraded was an explicitly set condition, use that. If not, use the deprecated failing.
+		degraded := failing
+		if degradedCondition != nil {
+			degraded = degraded_value
+		}
+
 		switch mode {
 		case resourcebuilder.InitializingMode:
-			// during initialization we allow failing as long as the component goes available
+			// during initialization we allow degraded as long as the component goes available
 			if available && (!progressing || len(expected.Status.Versions) > 0) {
 				return true, nil
 			}
 		default:
-			// if we're at the correct version, and available, and not failing, we are done
-			// if we're available, not failing, and not progressing, we're also done
+			// if we're at the correct version, and available, and not degraded, we are done
+			// if we're available, not degraded, and not progressing, we're also done
 			// TODO: remove progressing once all cluster operators report expected versions
-			if available && (!progressing || len(expected.Status.Versions) > 0) && !failing {
+			if available && (!progressing || len(expected.Status.Versions) > 0) && !degraded {
 				return true, nil
 			}
 		}
 
-		if c := resourcemerge.FindOperatorStatusCondition(actual.Status.Conditions, configv1.OperatorFailing); c != nil && c.Status == configv1.ConditionTrue {
+		condition := failingCondition
+		if degradedCondition != nil {
+			condition = degradedCondition
+		}
+		if condition != nil && condition.Status == configv1.ConditionTrue {
 			message := fmt.Sprintf("Cluster operator %s is reporting a failure", actual.Name)
-			if len(c.Message) > 0 {
-				message = fmt.Sprintf("Cluster operator %s is reporting a failure: %s", actual.Name, c.Message)
+			if len(condition.Message) > 0 {
+				message = fmt.Sprintf("Cluster operator %s is reporting a failure: %s", actual.Name, condition.Message)
 			}
 			lastErr = &payload.UpdateError{
 				Nested:  errors.New(lowerFirst(message)),
-				Reason:  "ClusterOperatorFailing",
+				Reason:  "ClusterOperatorDegraded",
```

Comment:
Did you change the places this is referenced elsewhere in the CVO?

Reply:
Doesn't look like it, change ClusterOperatorFailing in task.go and also change the message.

Reply:
fixed